src/rewrites.rs

```diff
@@ -114,6 +113,15 @@ pub fn rules_from_str(rs: Vec<&str>) -> Vec<Rewrite<Mdl, TensorAnalysis>> {
     rule_vec
 }
 
+pub fn pre_defined_rules() -> Vec<&'static str> {
```

Comment:
This doesn't need to be a function:
```
static PRE_DEFINE_RULES: &[&str] = [
    "(conv2d 1 1 0 0 ?input_1 ?input_2)=>(conv2d 1 1 0 0 ?input_1 (merge ?input_2 2))",
    "(conv2d 1 1 0 2 ?input_1 ?input_2)=>(conv2d 1 1 0 2 ?input_1 (merge ?input_2 2))",
    "(conv2d 2 2 0 0 ?input_1 ?input_2)=>(conv2d 2 2 0 0 ?input_1 (merge ?input_2 2))",
    "(conv2d 2 2 0 2 ?input_1 ?input_2)=>(conv2d 2 2 0 2 ?input_1 (merge ?input_2 2))",
];
```

Reply:
If I changed to static PRE_DEFINE_RULES: &[&str], I cannot do:

    let split_rules: Vec<&str> = rw_rules.split("\n").chain(PRE_DEFINED_RULES).collect();

Since the interator created for PRE_DEFINED_RULES will have item of type &&str instead of &str. I haven't found a nice way to both have the PRE_DEFINED_RULES as a static variable, and able to use chain to chain the iterators. Is there a neat way?

Reply:
You can use a map to remove a level of reference.
```
let pre_defined_rules = PRE_DEFINED_RULES.iter().map(|&x| x);
let split_rues: Vec<&str> = rw_rules.split("\n").chain(pre_defined_rules).collect();
```

Reply:
Thanks!